handlers/handlers.go

```diff
@@ -42,3 +48,177 @@ func Scheme(r *http.Request) string {
 	}
 	return "http"
 }
+
+type handlerFunc func(http.ResponseWriter, *http.Request) (int, error)
+
+func wrapGetWithErrors(ef func(error), hf handlerFunc) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			status := http.StatusMethodNotAllowed
+			http.Error(w, http.StatusText(status), status)
+			return
+		}
+		status, err := hf(w, r) // run the handlerFunc and obtain the return codes
+		if err != nil && ef != nil {
+			ef(err) // handle the error with the supplied function
+		}
+		// in case it's an error, write the status code for the requester
+		if status >= 400 {
+			http.Error(w, http.StatusText(status), status)
+		}
+	})
+}
+
+// ServiceInfo consists of two strings that contain the image type and a URL.
+type ServiceInfo struct {
+	ImageType string `json:"imageType"`
+	URL       string `json:"url"`
+}
+
+// ServiceSet is the base type for the HTTP handlers which combines multiple
+// mbtiles.DB tilesets.
+type ServiceSet struct {
+	tilesets map[string]*mbtiles.DB
+	Domain   string
+	Path     string
+}
+
+// New returns a new ServiceSet. Use AddDBFromPath to add a mbtiles file.
+func New() *ServiceSet {
+	return &ServiceSet{
+		tilesets: make(map[string]*mbtiles.DB),
+	}
+}
+
+// AddDB interprets filename a path to an mbtiles file which is opened and
+// served under id. In case the file cannot be opened, nil is returned.
```

Comment:
Also - this seems to only work with a path relative to the server.  What about merging this with `addDBWithSubPath` where `subpath` is empty by default.  In that case, the combination would become:
`func (s *ServiceSet) addDB(filename, path string) error`  (where `subpath` -> `path`)

Reply:
Thanks, the comment was indeed a copy & paste mistake. 

I hope I understood your change request right and have reworked `AddDB` (now `AddDBOnPath`) accordingly. I kept the method exported, though, otherwise a caller would be stuck after `New()`.